pkg/kubelet/server/stats/summary.go

```diff
@@ -40,7 +48,19 @@ var _ SummaryProvider = &summaryProviderImpl{}
 // NewSummaryProvider returns a SummaryProvider using the stats provided by the
 // specified statsProvider.
 func NewSummaryProvider(statsProvider StatsProvider) SummaryProvider {
-	return &summaryProviderImpl{statsProvider}
+	kubeletCreationTime := metav1.Now()
+	bootTime, err := util.GetBootTime()
+	if err != nil {
+		glog.Warningf("Error getting system boot time.  Node metrics will have an incorrect start time: %v", err)
+	}
+	// truncate bootTime to the second so it does not change when the kubelet restarts
```

Comment:
The uptime reported by the kernel is in seconds.  Time.Now() that we use as the kubelet start time has nanosecond precision.  We also collect the current time, and kernel uptime at different times, so there is likely a small difference between the measured "boot time" each time it is called because of nanosecond differences.

Reply:
I see. Maybe put this logic into GetBootTime()? I think it's natural for the caller to expect GetBootTime() to consistently return boot time with second precision.

Reply:
done